aten/src/ATen/FunctionalizeFallbackKernel.cpp

```diff
@@ -210,7 +210,13 @@ static at::Tensor lift_fresh_functionalize_copy(const at::Tensor & self) {
   // but that isn't really a use case today.
   // Needed for https://github.com/pytorch/pytorch/issues/105327
   if (at::functionalization::impl::isFunctionalTensor(self)) {
-    return self.clone();
+    // Note [Composite Functionalization under PreDispatch mode]
+    // When we are tracing under PreDispatch, PreDispatch key will be
+    // in the local include TLS. As a result, when we redispatch here,
+    // we will end up hitting PreDispatch stack first. So, we should
+    // directly redispatch to the functionalize key manually.
```

Comment:
nit: if we have a comment about why we call `._op_dk(DispatchKey.Functionalize)` from FunctionalTensorMode, it'd be good to link to that comment and mention the underlying reason is the same